### PR TITLE
Simplify Makefile

### PR DESCRIPTION
**Pull request by edif2008**
This PR contains a simplified version of the Makefile, as well as renaming from `output` to `dotnet`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,49 +11,41 @@ lib-win: client-win swig compile-win
...
 
 .PHONY: nupkg
 nupkg: lib lib-win
-	mkdir nuget
-	cp $(ODIR)/{Client.dll,libClient.so,Secret.cs,Client.cs,ClientPINVOKE.cs,SecretVersion.cs,secrethub.csproj} ./nuget/
-	dotnet pack nuget/secrethub.csproj
-	mv ./nuget/bin/Debug/SecretHub.*.nupkg .
-	rm -r ./nuget
-	rm $(ODIR)/libClient.so
+	dotnet pack $(ODIR)/secrethub.csproj -o $(ODIR)/build --nologo
```
Comment (Marton6) [inline Makefile]: What does the `--nologo` flag do?

File: Makefile [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
 SWIG_FILES = Client.cs secrethub_wrap.c ClientPINVOKE.cs Secret.cs SecretVersion.cs
-OUT_FILES = secrethub_wrap.o
-MONO_FILES = runme.exe
-DOTNET_FILES = secrethub secrethub.deps.json secrethub.dll secrethub.pdb secrethub.runtimeconfig.json
-DOTNET_DIRS = $(ODIR)/bin $(ODIR)/obj
-SWIG = swig
-CC = gcc
-ODIR = ./output
+OUT_FILES = secrethub_wrap.o libClient.so Client.dll
+ODIR = ./dotnet
```
Comment (Marton6) [inline Makefile]: Could we rename this to `DOTNET_DIR`?

File: Makefile [hunk 3]
```diff
@@ -16,49 +11,41 @@ lib-win: client-win swig compile-win
...
-# 	dotnet $(ODIR)/secrethub.dll
-
-.PHONY: mono-test
-mono: $(ODIR)/libClient.so
-	mono-csc -out:$(ODIR)/runme.exe $(ODIR)/*.cs
-# 	mono ./$(ODIR)/runme.exe
+dotnet-test: $(ODIR)/libClient.so
+	dotnet publish $(ODIR)/secrethub.csproj -o $(ODIR)/build -f netcoreapp3.1 --nologo
+	mv $(ODIR)/libClient.so $(ODIR)/build 
+# 	dotnet $(ODIR)/build/secrethub.dll
```
Comment (Marton6) [inline Makefile]: I assume this line will be uncommented when we have the tests ready, right?

**Review by Marton6 — COMMENTED** [review]
Good job simplifying the Makefile! It's much cleaner now. I have a couple questions/suggestions.

File: Makefile (hunk 3 above)
Comment (edif2008) [reply Makefile]: This line was for executing the `runme.cs` file. Based on the testing branch, this function will be modified. I will leave it as it is and modify it properly in the testing branch

File: Makefile (hunk 1 above)
Comment (edif2008) [reply Makefile]: It removes the dotnet banner that looks like this:
`Microsoft (R) Build Engine version 16.7.0+7fb82e5b2 for .NET`
`Copyright (C) Microsoft Corporation. All rights reserved.`

File: Makefile (hunk 2 above)
Comment (edif2008) [reply Makefile]: Fixed

File: Makefile (hunk 1 above)
Comment (Marton6) [reply Makefile]: Ah ok, then let's keep it

File: Makefile (hunk 3 above)
Comment (Marton6) [reply Makefile]: Okay

**Review by Marton6 — APPROVED** [review]
Looks good!

**Pull request merged**